lib/readyset/controller_extension.rb

```diff
@@ -0,0 +1,36 @@
+module Readyset
+  # Module providing controller extensions for routing ActiveRecord queries to a replica database.
+  module ControllerExtension
+    extend ActiveSupport::Concern
+
+    prepended do
+      # Routes ActiveRecord queries in specified actions to a replica database.
+      #
+      # This method defines an around_action callback that wraps
+      # the given actions. Inside the callback,
+      # it switches the ActiveRecord connection to a replica database,
+      # ensuring that all database queries
+      # within the actions are executed against the replica.
+      #
+      # @param actions [Array<Symbol, String>] A list of action names to which this routing applies.
+      # @param options [Hash] Additional options to refine the callback application,
+      # such as :only or :except.
+      # @param block [Proc] An optional block that can be used for additional
+      # logic around the action execution.
+      #
+      # @example
+      #   route_to_readyset :index, :show, only: [:index, :show] do |controller, action_block|
+      #     # Custom logic can be placed here
+      #   end
+      #
+      def self.route_to_readyset(*actions, **options, &block)
```

Comment:
Hmmm should we be passing the `&block` argument through to `around_action` here?

Reply:
There was an issue here, but it wasn't `&block`, I believe. 

The args are based off of [`_insert_callbacks`](https://apidock.com/rails/AbstractController/Callbacks/ClassMethods/_insert_callbacks), which contains an array (with an options hash at the end) and a `proc` that's added to the callbacks. 

You made a good point about not testing other people's code, so I modified the spec to just check that the args are being delegated correctly.